source/langdev/meta/lang/statix/debugging.rst

```diff
@@ -275,11 +283,48 @@ create the smallest set of rules and predicate arguments that still exhibit the
 debugging. A self-contained test is also very helpful when asking others for help, as it is much
 easier to review and run than having to setup and build a complete language project.
 
+.. _Common Problems:
+
 Some Common Problems
 --------------------
 
 - Predicates fail with ``amb(...)`` terms as arguments. These terms indicate parsing ambiguities,
   which should be fixed in the grammar (SDF3) files.
+- Errors in your specification appear at incorrect places (e.g. sort or constructor declarations).
+  In such cases, the declaration is referenced from an invalid position anywhere in your 
+  specification, but due to the non-deterministic order of constraint solving the error appears at
+  the observed position. The best approach to solve these issues is to comment away all usages,
+  until the error disappears. Then, in the last commented position, the declaration is used 
+  incorrectly.
+- One or both of the ``fileOk(...)`` or ``projectOk(...)`` predicates fail immediately, for example 
+  with the error messages:
+
+  .. code-block:: text
+
+     statics!fileOk(Scope("","s_1-1"),Test([Prog("A.mod",Decls(…)),Prog("B.mod",Decls(…)),Prog("C.mod",Decls(…))])) (no origin information)
+     statics!projectOk(Scope("","s_1-1")) (no origin information)
+ 
+  In such cases, you have probably moved the declarations of these predicates to another file.
+  
+  .. code-block:: statix
+  
+     // file: trans/mylang.stx
+     module mylang
+     imports mylang/program
+
+     rules
+
+       projectOk : scope
+       projectOk(s).
+  
+       fileOk : scope * Start   
+       fileOk(s, p) :- programOk(s, p).
+  
+  This is fine, but the call to the Statix solver should be updated accordingly. In 
+  ``trans/analysis.str``, the first term parameter to ``stx-editor-analyze`` (which is ``"statics"``
+  by default) should be updated to the new module name. 
```

Comment:
This should be refined a bit. How it works is that the first term argument is the module name of the spec to use, and the second is the name of the predicate to apply. Implicitly, the predicate name is prefixed by the module name (e.g., `statics!programOk`). I think the original top-level module `statics` is usually still where people collect all imports, so perhaps it is better to suggest to use a qualified name for the predicate.

Reply:
By "use a qualified name for the predicate", do you mean e.g. `"mylang!fileOk" as second argument? I don't think that would work, because it will still get (at least) a `"!"` prefixed.

Reply:
The predicate name only prefixed with the module name if it is unqualified. Or there is a bug.

Reply:
I'll rewrite this part a bit.